packages/flutter/test/material/outline_button_test.dart

```diff
@@ -44,78 +44,98 @@ void main() {
   });
 
 
-  testWidgets('Outline shape and border overrides', (WidgetTester tester) async {
-    debugDisableShadows = false;
-    const Color fillColor = Color(0xFF00FF00);
-    const Color borderColor = Color(0xFFFF0000);
-    const Color highlightedBorderColor = Color(0xFF0000FF);
+  testWidgets('OutlineButton shape and border component overrides', (WidgetTester tester) async {
+    const Color fillColor = const Color(0xFF00FF00);
+    const Color borderColor = const Color(0xFFFF0000);
+    const Color highlightedBorderColor = const Color(0xFF0000FF);
+    const Color disabledBorderColor = const Color(0xFFFF00FF);
     const double borderWidth = 4.0;
 
-    await tester.pumpWidget(
-      new Directionality(
+    Widget buildFrame(Key key, VoidCallback onPressed) {
+      return Directionality(
         textDirection: TextDirection.ltr,
         child: new Theme(
           data: new ThemeData(materialTapTargetSize: MaterialTapTargetSize.shrinkWrap),
           child: new Container(
             alignment: Alignment.topLeft,
-            child: new OutlineButton(
+            child: OutlineButton(
+              key: key,
               shape: const RoundedRectangleBorder(), // default border radius is 0
               color: fillColor,
               highlightedBorderColor: highlightedBorderColor,
+              disabledBorderColor: disabledBorderColor,
               borderSide: const BorderSide(
                 width: borderWidth,
                 color: borderColor,
               ),
-              onPressed: () { },
-              child: const Text('button')
+              onPressed: onPressed,
+              child: const Text('button'),
             ),
           ),
         ),
-      ),
-    );
-
-    final Finder outlineButton = find.byType(OutlineButton);
-    expect(tester.widget<OutlineButton>(outlineButton).enabled, true);
+      );
+    }
 
     final Rect clipRect = new Rect.fromLTRB(0.0, 0.0, 116.0, 36.0);
     final Path clipPath = new Path()..addRect(clipRect);
+
+    // Pump a widget with a null onPressed to make it disabled.
+    const Key disabledKey = Key('disabled');
+    await tester.pumpWidget(
+      buildFrame(disabledKey, null),
+    );
+
+    final Finder disabledOutlineButton = find.byKey(disabledKey);
+    expect(tester.widget<OutlineButton>(disabledOutlineButton).enabled, false);
+
     expect(
-      outlineButton,
+      disabledOutlineButton,
+      paints
+        ..clipPath(pathMatcher: coversSameAreaAs(clipPath, areaToCompare: clipRect.inflate(10.0)))
+        ..path(color: disabledBorderColor, strokeWidth: borderWidth));
+
+    // Pump a new widget with no onPressed to make it enabled.
```

Comment:
... with an onPressed callback, to make it enabled

Reply:
clarified here that it is no-op just to make it enabled, (as opposed to null which makes it disabled)